### PR TITLE
Fix issue 5554 soundness

### PR DESCRIPTION
**Pull request by olivier-aws**
### Description
This PR updates the Dafny CS runtime to use BigIntegers instead of int32 when retrieving the number of elements in a multiset. As this number may be over INT_MAX, this change avoids potential overflows. This is similar to what is done in the Java runtime.

### How has this been tested?
A test was added: Source/IntegrationTests/TestFiles/LitTests/LitTest/git-issues/git-issue-5554.dfy

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Source/IntegrationTests/TestFiles/LitTests/LitTest/git-issues/git-issue-5554.dfy [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+method Main() {
+    var initialMultiset := multiset{false};
+    
+    for iteration := 0 to 98 {
+        var multisetInstance := BooleanMultiset(initialMultiset); 
+        initialMultiset := initialMultiset + multisetInstance.containerSet;  
+    }
+    print initialMultiset > initialMultiset;
+}
```
Comment (typerSniper) [inline Source/IntegrationTests/TestFiles/LitTests/LitTest/git-issues/git-issue-5554.dfy:13]: needs new line at EOF

File: Source/IntegrationTests/TestFiles/LitTests/LitTest/git-issues/git-issue-5554.dfy.expect [hunk 2]
```diff
@@ -0,0 +1 @@
+false
```
Comment (typerSniper) [inline Source/IntegrationTests/TestFiles/LitTests/LitTest/git-issues/git-issue-5554.dfy.expect:1]: same here. I think the expect files are immune to trailing whitespaces, but I would double check.

File: Source/IntegrationTests/TestFiles/LitTests/LitTest/git-issues/git-issue-5554.dfy (hunk 1 above)
Comment (typerSniper) [reply Source/IntegrationTests/TestFiles/LitTests/LitTest/git-issues/git-issue-5554.dfy:13]: Btw, do you know why 98, or perhaps a better question is: is there a limit or does BigInteger support arbitrary sized integers?

Comment (olivier-aws) [reply Source/IntegrationTests/TestFiles/LitTests/LitTest/git-issues/git-issue-5554.dfy:13]: BigIntegers are unbounded, see https://learn.microsoft.com/en-us/dotnet/api/system.numerics.biginteger?view=net-8.0
In this example, the number of elements in the multiset after i iterations of the loop is 2^i. So there is an overflow in the computation of the size of the multiset (within the translation of the '>' operator in CS) as soon as the loop runs 31 times. Hence I change the 98 bound to 31.

**Review by typerSniper — APPROVED** [review]

**Pull request merged**